app/dao/users_dao.py

```diff
@@ -28,15 +28,17 @@ def save_user_attribute(usr, update_dict={}):
     db.session.commit()
 
 
-def save_model_user(usr, update_dict={}, pwd=None):
-    if pwd:
-        usr.password = pwd
-        usr.password_changed_at = datetime.utcnow()
+def save_model_user(user, update_dict={}, password=None, create_user=False):
```

Comment:
Optional: both this method and `update_user_password` could change the arguments you added to both be named the same thing for consistency e.g. `validate_email_access`

Reply:
I think I will leave that one.

Reply:
hmm, I thought about it some more, actually I like the idea, I'll change it.